x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/ApplicationPrivilege.java

```diff
@@ -43,96 +29,37 @@
  * The action patterns are entirely optional - many application will find that simple "privilege names" are sufficient, but
  * they allow applications to define high level abstract privileges that map to multiple low level capabilities.
  */
-public final class ApplicationPrivilege extends Privilege implements ToXContentObject, Writeable {
-
-    public static final String DOC_TYPE_VALUE = "application-privilege";
-
-    private static final ObjectParser<Builder, Boolean> PARSER = new ObjectParser<>(DOC_TYPE_VALUE, Builder::new);
-
-    static {
-        PARSER.declareString(Builder::applicationName, Fields.APPLICATION);
-        PARSER.declareString(Builder::privilegeName, Fields.NAME);
-        PARSER.declareStringArray(Builder::actions, Fields.ACTIONS);
-        PARSER.declareObject(Builder::metadata, (parser, context) -> parser.map(), Fields.METADATA);
-        PARSER.declareField((parser, builder, allowType) -> builder.type(parser.text(), allowType), Fields.TYPE,
-            ObjectParser.ValueType.STRING);
-    }
+public final class ApplicationPrivilege extends Privilege {
 
     private static final Pattern VALID_APPLICATION = Pattern.compile("^[a-z][A-Za-z0-9_-]{2,}$");
-    private static final Pattern VALID_APPLICATION_OR_WILDCARD = Pattern.compile("^[A-Za-z0-9_*-]+");
+    private static final Pattern VALID_APPLICATION_OR_WILDCARD = Pattern.compile("^[a-z*][A-Za-z0-9_*-]*");
     private static final Pattern VALID_NAME = Pattern.compile("^[a-z][a-zA-Z0-9_.-]*$");
+    private static final Pattern VALID_NAME_OR_ACTION = Pattern.compile("^\\p{Graph}*$");
```

Comment:
do you mind adding a comment about this? My regex isn't great so I had to go look up what this was